### ISSUE TITLE
RewritePatternRule does not support matrix parameters

### ISSUE THREAD

**Issue by celandro** [eclipse/jetty.project#430]
We ran into this and spent some time debugging it. It seems the correct place to fix it is in the applyURI rule where the query params are added back in. I am sure the other rewrite rules also have the same issue.

For now we are working around the issue with the following class which does have matrix params properly loaded in jax-rs2. I do not know if this code handles every possible matrix param case, but it handles the ones we need. To use just make a jar with this class file and put it in the lib/ext directory.

```
public class MatrixParamRewritePatternRule extends RewritePatternRule {
    public MatrixParamRewritePatternRule() {
        super();
    }
    @Override
    public void applyURI(Request request, String oldURI, String newURI) throws IOException
    {
        // hack in the matrix params
        int matrixParamIndex = oldURI.indexOf(';');
        if (matrixParamIndex > 0) {
            newURI = newURI + oldURI.substring(matrixParamIndex);
        }
        super.applyURI(request, oldURI, newURI);
    }

}
```

**Comment by joakime** [comment]
Are you talking about URI path parameters?

**Comment by celandro** [comment]
Yes, in Jax-rs its @MatrixParam

Example uri:
wget "http://localhost:7910/singles/qaas-v1/questionnaire/user/62116719/info;type=1;type=4?_=1457574072265"

**Comment by joakime** [comment]
Addressed in PR #4337 (Issue #4329)
